interfaces/builtin/bluez.go

```diff
@@ -207,21 +207,31 @@ func (iface *BluezInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *i
 	switch securitySystem {
 	case interfaces.SecurityAppArmor:
 		old := []byte("@SLOT_SECURITY_TAGS@")
-		buf := bytes.NewBuffer(nil)
-		fmt.Fprintf(buf, "snap.%s.{", slot.Snap.Name())
-		appNames := make([]string, 0, len(slot.Apps))
-		for appName := range slot.Apps {
-			appNames = append(appNames, appName)
-		}
-		sort.Strings(appNames)
-		for i, appName := range appNames {
-			if i > 0 {
-				fmt.Fprintf(buf, ",")
+		var new []byte
+		switch {
+		case len(slot.Apps) == 1:
+			for appName := range slot.Apps {
+				new = []byte(fmt.Sprintf("snap.%s.%s", slot.Snap.Name(), appName))
+			}
+		case len(slot.Apps) == len(slot.Snap.Apps):
+			new = []byte(fmt.Sprintf("snap.%s.*", slot.Snap.Name()))
+		case len(slot.Apps) != len(slot.Snap.Apps):
+			buf := bytes.NewBuffer(nil)
+			fmt.Fprintf(buf, "snap.%s.{", slot.Snap.Name())
+			appNames := make([]string, 0, len(slot.Apps))
+			for appName := range slot.Apps {
+				appNames = append(appNames, appName)
+			}
+			sort.Strings(appNames)
+			for i, appName := range appNames {
+				if i > 0 {
+					fmt.Fprintf(buf, ",")
+				}
+				fmt.Fprintf(buf, appName)
 			}
-			fmt.Fprintf(buf, appName)
+			fmt.Fprintf(buf, "}")
+			new = buf.Bytes()
 		}
```

Comment:
All this logic looks fine and I fully agree with the three cases. Personal preference would not be to use a case statement here since to my eyes it isn't as clear (if/else if/else would be clearer to me), but I won't block on that.